app/src/main/java/com/github/weg_li_android/MainActivity.kt

```diff
@@ -66,4 +265,163 @@ class MainActivity : AppCompatActivity() {
+                mode: ActionMode?,
+                menu: Menu?
+            ): Boolean {
+                return false
+            }
+
+            override fun onActionItemClicked(
+                mode: ActionMode,
+                item: MenuItem
+            ): Boolean {
+                return when (item.itemId) {
+                    R.id.delete_button -> {
+                        Timber.e("I'm here. Option1 selected, i will remove "+photoAdapter.deletePhotoSet.sorted().toString())
+                        photoAdapter.deletePhotoSet.sortedDescending().forEach {
+                            photoAdapter.removeAt(it)
+                        }
+                        photoAdapter.notifyDataSetChanged()
+
+                        mode.finish()
+                        true
+                    }
+                    else -> false
+                }
+            }
+
+            override fun onDestroyActionMode(mode: ActionMode) {
+                photoAdapter.deletePhotoSet.clear()
+                photoAdapter.notifyDataSetChanged()
+                mActionMode = null
+            }
+        }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (resultCode == Activity.RESULT_OK && data != null) {
+            when (requestCode) {
+                pickImage -> {
+                    if (data.clipData != null) {
+                        val count = data.clipData!!.itemCount
+                        for (i in 0 until count) {
+                            val imageUri: Uri? = data.clipData!!.getItemAt(i).uri
+                            val insertIndex = imageUri?.let { mainViewModel.addViolationPhoto(it) }
+                            if (insertIndex != null) {
+                                photoAdapter.notifyItemInserted(insertIndex)
+                            }
+                        }
+                    } else if (data.data != null) {
+                        val mImageUri: Uri? = data.data
+                        val insertIndex = mImageUri?.let { mainViewModel.addViolationPhoto(it) }
+                        if (insertIndex != null) {
+                            photoAdapter.notifyItemInserted(insertIndex)
+                        }
+                    }
+                }
+                takeImage -> {
+                    val bitmap = data.extras?.get("data") as Bitmap
+                    if (isStoragePermissionGranted()) {
+                        Timber.e("granted")
+                        saveAndInsertPhoto(bitmap)
+                    }
+                    else {
+                        tempBitmapOnRequestPermissions = bitmap
+                        Timber.e("Not granted")
+                    }
+                }
+            }
+        }
+    }
+
+    private fun saveAndInsertPhoto(bitmap : Bitmap) {
+        val imageUri : Uri = saveImageBitmap(bitmap, System.currentTimeMillis().toString())
```

Comment:
No need to specify the type here. The kotlin way is to let the IDE show the inferred type when possible.